### PR TITLE
Adding in support for mainnet

### PR DESCRIPTION
**Pull request by Pauan**
This PR allows users to use either testnet or mainnet (or both).

The user imports whichever network they want to use:

```js
import * as sdk from "@provablehq/sdk/testnet.js";
```

```js
import * as sdk from "@provablehq/sdk/mainnet.js";
```

### CODE REVIEW

File: wasm/build.js [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+            virtual({
+                "entry": `export { default } from "./Cargo.toml";`,
+            }),
+
+            rust({
+                cargoArgs: [
+                    // This enables multi-threading
+                    "--config", `build.rustflags=["-C", "target-feature=+atomics,+bulk-memory,+mutable-globals", "-C", "link-arg=--max-memory=4294967296"]`,
+                    "--no-default-features",
+                    "--features", `browser,${network}`,
```
Comment (iamalwaysuncomfortable) [inline wasm/build.js:34]: What are the consequences on this on node.js support?

File: wasm/src/types/networks/mod.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Aleo SDK library. If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(feature = "testnet")]
+mod testnet;
+
+#[cfg(feature = "testnet")]
+pub use testnet::*;
```
Comment (iamalwaysuncomfortable) [inline wasm/src/types/networks/mod.rs:21]: This is a good way to approach it.

File: wasm/build.js (hunk 1 above)
Comment (Pauan) [reply wasm/build.js:34]: There shouldn't be any consequences, everything should work the same in Node as it does right now.

This is just adding in a new `testnet` / `mainnet` feature flag, the `testnet` flag is identical to the current behavior.

**Comment by Pauan** [comment]
I added in some end-to-end tests under the `sdk/e2e` folder.

Right now I just copied the Node template, so it's just doing a basic "hello world".

But over time we can add more e2e tests and flesh them out.

**Comment by Pauan** [comment]
Also as discussed in Slack, I got rid of Jest and replaced it with Mocha + Chai + Sinon. The unit tests work so much better now.

**Comment by Pauan** [comment]
The PR is ready for review now, but it can't be merged until after mainnet launches, because it relies upon mainnet URLs that just don't exist right now.

After mainnet launches, we will re-run all of the tests until they're all green!

File: README.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <p align="center">
-    <a href="https://developer.aleo.org"> <img alt="Website" src="https://img.shields.io/badge/Developer_Docs-online-blue"></a>
+    <a href="https://developer.provable.com"> <img alt="Website" src="https://img.shields.io/badge/Developer_Docs-online-blue"></a>
```
Comment (jaketarnow) [inline README.md]: This can remain as `developer.aleo.org`

File: README.md [hunk 4]
```diff
@@ -61,9 +61,9 @@ Source: [Aleo Wasm](https://www.npmjs.com/package/@provablehq/wasm)
 
 ## 📚 Documentation
 
-#### [API Documentation](https://developer.aleo.org/sdk/typescript/overview)
+#### [API Documentation](https://developer.provable.com/sdk/typescript/overview)
```
Comment (jaketarnow) [inline README.md]: This can remain as `developer.aleo.org`

File: README.md [hunk 5]
```diff
@@ -61,9 +61,9 @@ Source: [Aleo Wasm](https://www.npmjs.com/package/@provablehq/wasm)
...
 API Documentation, tutorials for the Aleo SDK, and documentation on how to build Leo and Aleo Instructions programs can
-be found on the [Aleo Developer Docs](https://developer.aleo.org/sdk/typescript/overview) page.
+be found on the [Aleo Developer Docs](https://developer.provable.com/sdk/typescript/overview) page.
```
Comment (jaketarnow) [inline README.md]: This can remain as `developer.aleo.org`

File: website/src/pages/TermsOfUse.jsx [hunk 6]
```diff
@@ -16,7 +16,7 @@ const TermsOfUse = () => {
                 provided by Provable Inc., a Delaware corporation (including its
```
Comment (jaketarnow) [inline website/src/pages/TermsOfUse.jsx]: Do not update anything on `/website/src/pages/TermsOfUse`

File: create-leo-app/template-react-leo/README.md [hunk 7]
```diff
@@ -40,12 +40,12 @@ Your app should be running on http://localhost:5173/
...
 ## Deploy program from web app
 
-> [!WARNING]  
+> [!WARNING]
 > This is for demonstration purposes or local testing only, in production applications you
 > should avoid building a public facing web app with private key information
 
 Information on generating a private key, seeding a wallet with funds, and finding a spendable record can be found here
-if you are unfamiliar: https://developer.aleo.org/testnet/getting_started/deploy_execute_demo
+if you are unfamiliar: https://developer.provable.com/testnet/getting_started/deploy_execute_demo
```
Comment (jaketarnow) [inline create-leo-app/template-react-leo/README.md]: This can remain as `developer.aleo.org`

File: create-leo-app/template-react-ts/README.md [hunk 8]
```diff
@@ -40,12 +40,12 @@ Your app should be running on http://localhost:5173/
...
 ## Deploy program from web app
 
-> [!WARNING]  
+> [!WARNING]
 > This is for demonstration purposes or local testing only, in production applications you
 > should avoid building a public facing web app with private key information
 
 Information on generating a private key, seeding a wallet with funds, and finding a spendable record can be found here
-if you are unfamiliar: https://developer.aleo.org/testnet/getting_started/deploy_execute_demo
+if you are unfamiliar: https://developer.provable.com/testnet/getting_started/deploy_execute_demo
```
Comment (jaketarnow) [inline create-leo-app/template-react-ts/README.md]: This can remain as `developer.aleo.org`

File: sdk/README.md [hunk 9]
```diff
@@ -4,7 +4,7 @@
 </p>
 
 <p align="center">
-    <a href="https://developer.aleo.org"> <img alt="Website" src="https://img.shields.io/badge/Developer_Docs-online-blue"></a>
+    <a href="https://developer.provable.com"> <img alt="Website" src="https://img.shields.io/badge/Developer_Docs-online-blue"></a>
```
Comment (jaketarnow) [inline sdk/README.md]: This can remain as `developer.aleo.org`

File: sdk/README.md [hunk 10]
```diff
@@ -182,7 +182,7 @@ function hello:
 The SDK provides the ability to execute Aleo programs 100% client-side within the browser.
 
 The `ProgramManager` object encapsulates the functionality for executing programs and making zero-knowledge proofs about
-them. Under the hood it uses cryptographic code compiled from [SnarkVM](https://developer.aleo.org/aleo) into WebAssembly
+them. Under the hood it uses cryptographic code compiled from [SnarkVM](https://developer.provable.com/aleo) into WebAssembly
```
Comment (jaketarnow) [inline sdk/README.md]: This can remain as `developer.aleo.org`

File: wasm/src/lib.rs [hunk 11]
```diff
@@ -41,7 +41,7 @@
 //! * Aleo primitives such as `Records`, `Programs`, and `Transactions` and their associated helper methods
 //! * A `ProgramManager` object that contains methods for authoring, deploying, and interacting with Aleo programs
 //!
-//! More information on these concepts can be found at the [Aleo Developer Hub](https://developer.aleo.org/concepts).
+//! More information on these concepts can be found at the [Aleo Developer Hub](https://developer.provable.com/concepts).
```
Comment (jaketarnow) [inline wasm/src/lib.rs]: Use `developer.aleo.org`

File: website/src/pages/Homepage.jsx [hunk 12]
```diff
@@ -32,14 +32,14 @@ const Homepage = () => {
                 <Link
                     target="_blank"
                     rel="noopener noreferrer"
-                    to="https://developer.aleo.org/sdk/"
+                    to="https://developer.provable.com/sdk/"
```
Comment (jaketarnow) [inline website/src/pages/Homepage.jsx]: Use `developer.aleo.org`

File: .github/ISSUE_TEMPLATE/bug.md [hunk 13]
```diff
@@ -10,8 +10,8 @@ labels: bug
 <!--
     What's the bug in the Aleo SDK that you found?
     How serious is this bug and what is affected?
-    
-    To report a security issue in the Aleo SDK, please email security@aleo.org.
+
```
Comment (jaketarnow) [inline .github/ISSUE_TEMPLATE/bug.md:13]: Keep as `security@aleo.org`

**Review by jaketarnow — CHANGES_REQUESTED** [review]

**Comment by iamalwaysuncomfortable** [comment]
One final comment from my end is to update all SnarkVM dependencies to rev `3d42aa0` (i.e. the latest stable rev)

**Review by iamalwaysuncomfortable — COMMENTED** [review]
One final comment from my end is to update all SnarkVM dependencies to rev `3d42aa0` (i.e. the latest stable rev). This ensures the correct `Network` trait constants are present in the SDK!

**Review by jaketarnow — APPROVED** [review]
LGTM

**Pull request merged**